qemu/tests/rng_hotplug.py

```diff
@@ -123,3 +136,25 @@ def stop_rngd(vm):
         time.sleep(5)
         for dev in dev_list:
             unplug_rng(vm, dev)
+
+    new_dev = qdevices.QDevice("virtio-rng-pci",
+                               {'id': 'virtio-rng-pci-0'})
+    hotplug_rng(vm, new_dev)
+
+    # Run test after hotplug
+    if test_after_hotplug:
+        restart_rngd(vm)
+        error_context.context("Run %s after hotplug" % test_after_hotplug)
+        utils_test.run_virt_sub_test(test, params, env, test_after_hotplug)
+
+    if sub_type_after_plug:
+        error_context.context("Running sub test %s after plug" % sub_type_after_plug, logging.info)
+        utils_test.run_virt_sub_test(test, params, env, sub_type_after_plug)
+
+        if sub_type_after_plug == "shutdown":
+            return
+
+        if test_after_hotplug:
+            restart_rngd(vm)
+            error_context.context("Run %s after %s" % (test_after_hotplug, sub_type_after_plug))
+            utils_test.run_virt_sub_test(test, params, env, test_after_hotplug)
```

Comment:
Seems a few duplicate code above. I'm not sure if there is a way to reduce it. 
And it is a little strange to move the first hotunplug and the last hotplug out of the "for" loop.

Reply:
Check rng function after device hotplug and after guest reboot is a necessary step for our cases,the check step is already encapsulated in a function.The duplicated code you mention is just because we need these steps.

About the for loop,I move the first and last plug out of it just because it makes the code more pretty, else I need to add a few lines to check the judging condition and put all the test_after_hotplug/sub_type_after_plug/sub_type_after_unplug test all in the loop,it will make the for loop too big and chaos.
As I actually don't need to do those tests during the loop,I'd rather separate them.

@huangyum What do you think?

Reply:
I will check if I can encapsulated the following check steps into a new function:
 if test_after_hotplug:
            restart_rngd(vm)
            error_context.context("Run %s after %s" % (test_after_hotplug, sub_type_after_plug))
            utils_test.run_virt_sub_test(test, params, env, test_after_hotplug)

Thank you for your advice